### DISCUSSION TITLE
Nextcloud-AIO reports 127.0.0.1 as remote address

### DISCUSSION THREAD

**Discussion by DragonPi** [nextcloud/all-in-one#5218] [Questions] [answered]
Hello,

I'm having difficulties with a throttled 127.0.0.1 address.

Originally the address reported was '172.23.0.1' So I added it as per: https://github.com/nextcloud/all-in-one/discussions/2045 as a trusted proxy. Since then the address reported is 127.0.0.1

```php
  'trusted_proxies' =>
  array (
    0 => '127.0.0.1',
    1 => '::1',
    10 => '172.18.0.0/16',
    2 => '172.23.0.1',
  ),
```

I also activated/configured the IPv6 as mentioned in the same thread. But to no avail.

I have 3 RPi's involved in the setup:
1) hosts nginx proxy manager (config from https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md)
2) hosts DNS, DHCP etc... for name resolving
3) hosts the nextcloud-aio as docker rootless setup

No idea at the moment on how to proceed.

Any thoughts?

regards,
Koen

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker info` here?

**Comment by DragonPi** [comment]
Here you go:

```text
Client:
 Version:    27.1.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 9
  Running: 8
  Paused: 0
  Stopped: 1
 Images: 10
 Server Version: 27.1.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  rootless
  cgroupns
 Kernel Version: 6.6.31+rpt-rpi-v8
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.669GiB
 Name: diomedes
 ID: bec1a71f-2d7d-4103-87b9-9e0106fc227d
 Docker Root Dir: /home/aio/.local/share/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No cpuset support
WARNING: No io.weight support
WARNING: No io.weight (per device) support
WARNING: No io.max (rbps) support
WARNING: No io.max (wbps) support
WARNING: No io.max (riops) support
WARNING: No io.max (wiops) support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Reply by szaimen** [reply, marked as answer]
Does this help?
https://github.com/nextcloud/all-in-one/blob/main/docker-rootless.md#note-regarding-docker-network-driver

**Comment by DragonPi** [comment]
Thanks @szaimen for reaching out so fast.

Added the override.conf as mentioned here: https://rootlesscontaine.rs/getting-started/docker/#changing-the-port-forwarder

I now get the public IP of the host where Nginx Proxy Manager is running, which is a step forward, but I guess something must be wrong then in the config of the reverse proxy as I expect to see the client's address. Don't know I you have experience with that, otherwise I'll reach for the NPM forum and search additional help there.

**Comment by DragonPi** [comment]
Additional question: Do I need to add the IP from the machine hosting the NPM in the trusted proxies also?

**Reply by szaimen** [reply]
Yes

**Comment by DragonPi** [comment]
Still no success.

Added:
`proxy_set_header X-Real-IP $remote_addr;` as per https://github.com/NginxProxyManager/nginx-proxy-manager/discussions/3215
in advanced settings from the proxy host in NPM. But this doesn't change anything. Moreover, as I understood correctly: this setting is default, so adding it would not be necessary.

However: since I added the IP from the NPM host in `trusted_proxies` the reported IP is again `127.0.0.1` ? Is that behavior correct?

regards,
Koen
